crates/gossip/src/harness.rs

```diff
@@ -322,13 +333,20 @@ pub struct K2GossipFunctionalTestFactory {
     space_id: SpaceId,
     /// The builder for creating new instances.
     builder: Arc<Builder>,
+    /// A sender for ops stored events.
+    ///
+    /// This must not be used for sending. If it is provided then the K2 mem op store is in use
+    /// and that will send ops through this channel. Use the sender to create new listeners so
+    /// that each space can update its DHT.
+    ops_stored_sender: Option<tokio::sync::broadcast::Sender<Vec<StoredOp>>>,
```

Comment:
This is new and was a big part of why the test wasn't doing what it was supposed to do. When we store ops, we need to inform the DHT model. This most of the changes in the rest of this file are about making that happen